Fix: Add validation for invalid "reset-" option keys in config files (#2083)
f3d-app/f3d: application/testing/CMakeLists.txt

# Test invalid config file
f3d_test(NAME TestInvalidConfigFile DATA cow.vtp CONFIG ${F3D_SOURCE_DIR}/testing/configs/invalid.json REGEXP "Unable to parse the configuration file" NO_BASELINE)

# Test invalid reset key in config file

> f3d_test(NAME TestInvalidResetKey DATA cow.vtp CONFIG ${F3D_SOURCE_DIR}/testing/configs/invalid_reset.json REGEXP "Invalid option: 'reset-' must be followed by a valid option name, ignoring entry" NO_BASELINE)

Reviewer: instead of relying on a config file, it would be better to test with `--reset-` in the command line.

```suggestion
f3d_test(NAME TestInvalidResetOptions DATA cow.vtp ARGS `--reset-` REGEXP "Invalid option: 'reset-' must be followed by a valid option name, ignoring entry" NO_BASELINE)
```

This way you can drop the config file.

Author: I reworked the test to use --reset- as suggested, but it doesn’t reach UpdateOptions() — it's rejected earlier by the CLI parser. So either we drop the test or keep it config-based. Let me know what works best for you!

> 

Reviewer: you can remove this line

# Test invalid config file
f3d_test(NAME TestInvalidConfigFile DATA cow.vtp CONFIG ${F3D_SOURCE_DIR}/testing/configs/invalid.json REGEXP "Unable to parse the configuration file" NO_BASELINE)

# Test invalid reset key in config file

> f3d_test(NAME TestInvalidResetOptions DATA cow.vtp ARGS `--reset=` REGEXP "Invalid option: 'reset' must be followed by a valid option name, ignoring entry" NO_BASELINE)

Reviewer: ```suggestion
f3d_test(NAME TestInvalidResetOptions DATA cow.vtp ARGS --reset= REGEXP "Invalid option: 'reset' must be followed by a valid option name, ignoring entry" NO_BASELINE)
```

> f3d_test(NAME TestInvalidResetOptions DATA cow.vtp ARGS `--reset-` REGEXP "Invalid option: 'reset-' must be followed by a valid option name, ignoring entry" NO_BASELINE)

Reviewer: ```suggestion
f3d_test(NAME TestInvalidResetOptions DATA cow.vtp ARGS `--reset=` REGEXP "Invalid option: 'reset-' must be followed by a valid option name, ignoring entry" NO_BASELINE)
```

Author: The same result

Reviewer: not the same. You can use `-VV` to look at the command line and figure out how to fix it :)
